apps/issuer/admin.py

```diff
@@ -8,10 +8,86 @@
 from mainsite.admin import badgr_admin
 
 from .models import Issuer, BadgeClass, BadgeInstance, BadgeInstanceEvidence, BadgeClassAlignment, BadgeClassTag, \
-    BadgeClassExtension, IssuerExtension, BadgeInstanceExtension
+    BadgeClassExtension, IssuerExtension, BadgeInstanceExtension, SuperBadge, CollectionBadgeContainer
 from .tasks import resend_notifications
 
 
+class CollectionBadgeInstanceInline(TabularInline):
+    model = CollectionBadgeContainer.assertions.through
+    extra = 0
+
+class CollectionBadgeAdmin(ModelAdmin):
+    readonly_fields = ('created_by', 'created_at', 'updated_at', 
+                       'entity_id')
+    list_display = ('name', 'entity_id')
+    list_display_links = ('name',)
+    list_filter = ('created_at',)
+    search_fields = ('name', 'entity_id')
+    fieldsets = (
+        ('Metadata', {
+            'fields': ('created_by', 'created_at', 'updated_at', 'entity_id'),
+            'classes': ("collapse",)
+        }),
+        (None, {
+            'fields': ('image', 'name', 'description')
+        }),
+    )
+    inlines = [
+        CollectionBadgeInstanceInline,
+    ]
+
+    def save_model(self, request, obj, form, change):
+        force_resize = False
+        if 'image' in form.changed_data:
+            force_resize = True
+        obj.save(force_resize=force_resize)
+
+    def badge_image(self, obj):
+        return mark_safe('<img src="{}" width="32"/>'.format(obj.image.url)) if obj.image else ''
+    badge_image.short_description = 'Badge'
+    
+    pass
+
+
+
+class SuperBadgeInstanceInline(TabularInline):
+    model = SuperBadge.assertions.through
+    extra = 0
+    # raw_id_fields = ('badgeinstance',)
+
+class SuperBadgeAdmin(ModelAdmin):
```

Comment:
Isn't this exactly the same as `CollectionBadgeAdmin`? Maybe the duplicate code can be extracted into a super class?

Reply:
I dont necessarily expect the classes to stay the same. I think at the moment it makes sense to keep them seperate, but if they dont diverge later on i will create a super class.